nixos/modules/services/audio/mpd.nix

```diff
@@ -126,49 +117,136 @@ in
         '';
       };
 
-      user = lib.mkOption {
-        type = lib.types.str;
-        default = name;
-        description = "User account under which MPD runs.";
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = lib.elem cfg.settings.bind_to_address [
+          "any"
+          "0.0.0.0"
+        ];
+        defaultText = lib.literalExpression ''
+          elem config.services.mpd.settings.bind_to_address ["any" "0.0.0.0"]
+        '';
+        description = "Open ports in the firewall for mpd.";
       };
 
-      group = lib.mkOption {
-        type = lib.types.str;
-        default = name;
-        description = "Group account under which MPD runs.";
-      };
+      settings = lib.mkOption {
+        type = lib.types.submodule {
+          freeformType =
+            with lib.types;
+            oneOf [
+              str
+              int
+              bool
+              path
+            ];
+          options = {
+            music_directory = lib.mkOption {
+              type = with lib.types; either path (strMatching "(http|https|nfs|smb)://.+");
```

Comment:
I believe what schemas are valid depends on what storage plugins you have activated. There is at least `udisks://` as well. Maybe it would be better to just ensure that it looks like a URI somehow, rather than hardcode the schemas